src/selector.js

```diff
@@ -209,8 +193,7 @@ function find( selector, context, results, seed ) {
 			}
 
 			// Take advantage of querySelectorAll
-			if ( !nonnativeSelectorCache[ selector + " " ] &&
-				( !rbuggyQSA || !rbuggyQSA.test( selector ) ) ) {
```

Comment:
At this point, all buggy qSA handling applied to IE only. With this gone, perhaps we wouldn’t need rewriting of buggy selectors in 4.0. We’d still need parsing to inject `:scope` or a temporary ID and to support custom pseudos, though.

Reply:
This code is back here as we had to handle `:has`.